ambry-messageformat/src/main/java/com.github.ambry.messageformat/MessageFormatRecord.java

```diff
@@ -171,6 +171,44 @@ static boolean isValidBlobRecordVersion(short blobRecordVersion) {
   }
 
   /**
+   * Deserialize a complete blob record into a {@link BlobAll} object.
+   * @param stream the {@link InputStream} from which to read the blob record.
+   * @param storeKeyFactory the factory for parsing store keys.
+   * @return a {@link BlobAll} object with the {@link BlobInfo} and {@link BlobData} for the blob.
+   * @throws IOException
+   * @throws MessageFormatException
+   */
+  public static BlobAll deserializeBlobAll(InputStream stream, StoreKeyFactory storeKeyFactory)
+      throws IOException, MessageFormatException {
+    skipHeaders(stream);
```

Comment:
are you sure this is the right thing to do? The header has versions and other useful things.

Take a look at `MessageSievingInputStream::checkForMessageValidity()` which deserializes the whole record. You want to do something similar.